api/tests/LinkyLink.Integration.Tests/LinkApiTests.cs

```diff
@@ -0,0 +1,135 @@
+using LinkyLink.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Identity.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace LinkyLink.Integration.Tests
+{
+    [TestClass]
+    public class LinkApiTests : IntegrationTest, IDisposable
+    {
+        private readonly HttpClient client;
+        private readonly AzureAdB2C b2cConfig;
+
+        public LinkApiTests()
+        {
+            this.b2cConfig = this.Configuration.GetSection("AzureAdB2C").Get<AzureAdB2C>();
+
+            this.client = new HttpClient
+            {
+                BaseAddress = new Uri(this.b2cConfig.BaseAddress)
+            };
+        }
+
+        private async Task<string> CreateAccessTokenAsync()
+        {
+            var app = PublicClientApplicationBuilder
+                .Create(this.b2cConfig.ClientId)
+                .WithDefaultRedirectUri()
+                .WithB2CAuthority(this.b2cConfig.Authority)
+                .Build();
+
+            var credentials = new NetworkCredential(this.b2cConfig.Username, this.b2cConfig.Password);
+
+            var appScope = new List<string>(this.b2cConfig.Scope.Split(' '));
+
+            var authResult = await app
+                .AcquireTokenByUsernamePassword(appScope, credentials.UserName, credentials.SecurePassword)
+                .ExecuteAsync();
+
+            return authResult.AccessToken;
+        }
+
+        [TestMethod]
+        public async Task Links_API_CRUD_Operations()

```

Comment:
Can we break this test into multiple methods  where each performs a certain part of the test and then call all of them in sequence within this test?

Reply:
Yes - I almost did the same :)

Reply:
great minds think a like :sunglasses: